Sofa/framework/Core/src/sofa/core/CollisionElement.h

```diff
@@ -264,13 +264,13 @@ class TCollisionElementIterator : public BaseCollisionElementIterator
     }
 
     /// Distance to the actual (visual) surface
-    double getProximity() { return model->getProximity(); }
+    [[nodiscard]] double getProximity() const { return model->getProximity(); }
```

Comment:
Do you know why those functions return a `double` and not a `SReal`?
Since you change the signature, maybe we can also change the returned type.

Reply:
I suppose it is double because.... just because :p
Actually if we change it by SReal it will only change the signature if compiling in float only (which not a lot of people bother, as the compilation regularly breaks when selecting float...)
I will change it then, for the sake of consistency

Reply:
And the other file CollisionModel uses SReal by the way...